vignettes/available-theming-variables.Rmd

```diff
@@ -0,0 +1,258 @@
+---
+title: "Available theming variables"
+output: rmarkdown::html_vignette
+author: "Nick Strayer"
+date: "`r Sys.Date()`"
+vignette: >
+  %\VignetteIndexEntry{available-theming-variables}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  echo = FALSE, 
+  message = FALSE
+)
+
+htmltools::tagList(
+  rmarkdown::html_dependency_font_awesome(),
+  rmarkdown::html_dependency_bootstrap("default")
+)
+```
+
+```{r setup}
+library(dplyr)
+library(stringr)
+library(purrr)
+library(glue)
+library(bslib)
+```
+
+```{css}
+table.dataTable tr[role="row"] td {
+  font-family: monospace;
+}
+table.dataTable tr[role="row"] td:nth-child(3) {
+  font-family: inherit;
+  font-size: 0.85rem;
+}
+table.dataTable tr[role="row"] td:nth-child(4) {
+  text-align: center;
+}
+.dep-link {
+  color:steelblue; 
+  text-decoration:underline;
+  cursor: pointer;
+}
+.selected .dep-link {
+  color: white;
+}
+```
+
+```{r variables_df}
+# BS3
+# Not using BS3 now because the comment templates are different so a whole new scraping pipeline is needed
+# sass_variables_loc <- "https://raw.githubusercontent.com/rstudio/bslib/master/inst/lib/bs-sass/assets/stylesheets/bootstrap/_variables.scss"
+
+# BS4
+sass_variables_loc <- rprojroot::find_package_root_file("inst/lib/bs/scss/_variables.scss")
+sass_variables_url <- file.path(
+  "https://github.com/rstudio/bslib/blob", 
+   substr(system("git rev-parse HEAD", intern = TRUE), 1, 7),
+  "inst/lib/bs/scss/_variables.scss"
+)
+# A section is denoted by a comment followed by either an empty line or an empty comment.
+# e.g.
+# // Section ID
+#
+# Or
+# // Section ID
+# //
+#
+# Sometimes the sections are just isolated floating comments
+#
+# // Section ID
```

Comment:
This worries me a little bit, maybe we should patch the source file so that all the sections adhere to:

```
// Section title
//
// Section description
```

Reply:
This way, we'll have some sort of guarantee that we're not creating sections that we're not aware of

Reply:
Done here: https://github.com/rstudio/bslib/pull/238/commits/2256910d1a7cdd56f7aeb37d32f6a782b845fd72